### PR TITLE
Provider Changes for Cluster Deployment

### PR DESCRIPTION
**Pull request by Krunal-Thakkar**
# Description
Handle the Client, For Initial Client connection and Block API Client Connection.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration
image

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: powerflex/provider/provider.go [hunk 1]
```diff
@@ -41,7 +41,16 @@ func New() provider.Provider {
...
 
-type powerflexProvider struct{}
+type powerflexProvider struct {
+	client        *goscaleio.Client
+	clientError   string
+	gatewayClient *goscaleio.GatewayClient
+}
+
+// type powerflexClients struct {
+// 	client        *goscaleio.Client
```
Comment (shenda1) [inline powerflex/provider/provider.go:51]: Pls remove it if not required

Comment (Krunal-Thakkar) [reply powerflex/provider/provider.go]: Removed.

**Review by shenda1 — DISMISSED** [review]
LGTM. Please attach the acceptance tests results

File: powerflex/provider/provider.go [hunk 2]
```diff
@@ -99,6 +103,7 @@ func (p *powerflexProvider) Schema(_ context.Context, _ provider.SchemaRequest,
 func (p *powerflexProvider) Configure(ctx context.Context, req provider.ConfigureRequest, resp *provider.ConfigureResponse) {
 	tflog.Info(ctx, "Configuring powerflex client")
 
+	// var pClients powerflexClients
```
Comment (AnikaAgiwal2711) [inline powerflex/provider/provider.go:106]: commented code can be removed

**Comment by AnikaAgiwal2711** [comment]
Rest looks good

**Review by AnikaAgiwal2711 — APPROVED** [review]

File: powerflex/provider/provider.go (hunk 2 above)
Comment (Krunal-Thakkar) [reply powerflex/provider/provider.go]: removed.

**Review by shenda1 — APPROVED** [review]

**Pull request merged**